src/main/kotlin/org/jqassistant/tooling/intellij/plugin/editor/templates/RulejQATemplateFileCreation.kt

```diff
@@ -0,0 +1,164 @@
+        val project = e.project ?: return
+        val virtualFile = e.getData(com.intellij.openapi.actionSystem.CommonDataKeys.VIRTUAL_FILE) ?: return
+        val directoryPath = virtualFile.path
+
+        // Custom name prompt
+        val customName = Messages.showInputDialog(
+            project,
+            "Enter the name for the Rule XML file (without .xml):",
+            "Custom Rules File",
+            Messages.getQuestionIcon()
+        )?.trim() // trim() removes leading and trailing spaces
+
+        // Verify that the user provided valid input
+        if (customName.isNullOrEmpty()) {
+            Messages.showWarningDialog(
+                project,
+                "The file name cannot be empty.",
+                "Invalid Input",
+            )
+            return // Cancel if input is empty
+        }
+        // Check if only acceptable Characters are used for file name
+        if(!isNameValid(customName)) {
+            Messages.showWarningDialog(
+                project,
+                "The file name cannot be contaminated with abstract Characters",
+                "Invalid Input",
+            )
+            return // Cancel if input is not supported
+        }
+        // File name with the user-defined name and the extension .xml
+        val targetFile = File(directoryPath, "$customName.xml")
+
+        // Check whether the file already exists
+        if (targetFile.exists()) {
+            Messages.showErrorDialog(
+                project,
+                "A file named $customName.xml already exists in $directoryPath",
+                "File Exists",
+            )
+            openFileAutomatic(project, targetFile)
+            return
+        }
+
+        // Load template from resources and insert the name
+        val resourceStream = javaClass.getResourceAsStream("/templates/my_rules.xml")
+        if (resourceStream != null) {
+            val content = resourceStream.bufferedReader().use { it.readText() }
+
+            // Replace placeholders in content with custom name
+            val modifiedContent = content.replace("{{CUSTOM_NAME}}", customName)
+
+            // Save file with the modified content in the target directory
+            targetFile.writeText(modifiedContent)
+
+            Messages.showInfoMessage(
+                project,
+                "$customName.xml was successfully added to $directoryPath",
+                "File Created",
+            )
+            // Update project structure for new files
+            VirtualFileManager.getInstance().syncRefresh()
+            openFileAutomatic(project, targetFile)
+        } else {
+            Messages.showErrorDialog(
+                project,
+                "Template file my_rules.xml not found in plugin resources.",
+                "Error",
+            )
+        }
+    }
+
+    fun isNameValid(name: String) : Boolean{
+        //All supported Characters of a file name
+        val validNameRegex = Regex("^[a-zA-Z0-9._-]+$")
```

Comment:
```suggestion
        val validNameRegex = Regex("^[a-zA-Z0-9._\-]+$")
```
The original works at the moment, but if someone extends the regex it would turn into a char range.